File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l=0;
        int p=0;

        for(int r=1;r<prices.size();r++){
            p=max(p,prices[r]-prices[l]);
            if(prices[l]>prices[r]){
                l=r;
            }
        }
        return p;

        
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_count=0
        seen={}
        l=0
        for r in range(len(s)):
            if s[r] in seen:
                l=max(l,seen[s[r]]+1)
            seen[s[r]]=r
            max_count=max(max_count,r-l+1)
        return max_count


        